test.sh

```diff
@@ -182,7 +175,15 @@ fi
 # for all boulder packages
 if [ -z "${UNIT_PACKAGES[@]+x}" ]
 then
-  UNIT_PACKAGES+=("-p" "1" "./...")
+  # '-p=1' configures unit tests to run serially, rather than in parallel. Our
+  # unit tests depend on mutating a database and then cleaning up after
+  # themselves. If these test were run in parallel, they could fail spuriously
+  # due to one test modifying a table (especially registrations) while another
+  # test is reading from it.
+  # https://github.com/letsencrypt/boulder/issues/1499
+  # https://pkg.go.dev/cmd/go#hdr-Testing_flags
+  UNIT_FLAGS+=("-p=1")
```

Comment:
Honestly I wonder how many of our tests actually fail if we remove this. I think a lot of the tests could be modified to (or maybe already do?) create and throw away a fresh in-memory database rather than modifying and clearing a shared in-memory database. Obviously way out of scope for this PR but it would be nice to be able to remove this in the mid-term future.